livingatlas/scripts/la-pipelines

```diff
@@ -0,0 +1,621 @@
+
+CMD=$(basename $0)
+
+source ./set-env.sh
+
+FIND_DOCOPTS=$(which docopts)
+FIND_YQ=$(which yq)
+
+if [[ -z $FIND_DOCOPTS ]]
+then
+    echo "ERROR: Please install docopts https://github.com/docopt/docopts an copy it in your PATH"
+    exit 1
+fi
+
+if [[ -z $FIND_YQ ]]
+then
+    echo "ERROR: Please install yq, see https://github.com/mikefarah/yq"
+    exit 1
+fi
+
+set -e # Stop on any failure
+
+# Detect if we are in production or not
+if [[ $PWD == "/usr/bin" ]] ; then PROD=true ; else PROD=false ; fi
+
+if [[ $PROD = false ]]; then
+    VER=$(grep -oPm1 "(?<=<version>)[^<]+" "../pom.xml")
+else
+    VER=$(dpkg-query --show -f '${Version}' la-pipelines)
+fi
+
+WHEREL="[--local|--embedded|--cluster]"
+WHERE="[--embedded|--cluster]"
+
+eval "$(docopts -V - -h - : "$@" <<EOF
+
+LA-Pipelines data ingress utility.
+
+The $CMD can be executed to run all the ingress steps or only a few of them:
+
+Pipeline ingress steps:
+
+    ┌───── do-all ──────────────────────────────────┐
+    │                                               │
+dwca-avro --> interpret --> uuid -->                │
+     export-latlng --> sample --> sample-avro --> index
+
+-  'dwca-avro' reads a Dwc-A and converts it to an Avro file;
+-  'interpret' reads a Dwc-A, interprets it and write the interpreted data and the issues in Avro files;
+-  'uuid' minting on new records and rematching existing UUIDs to records that have been previously loaded;
+-  'export-latlng' exports a unique set of coordinates for a dataset into CSV;
+-  'sample' crawl the LA layers. Requires an input csv containing lat, lng (no header) and output directory.
+-  'sample-avro' adds a sampling AVRO extension to the stored interpretation
+-  'index' reads a Dwc-A, interprets it and index the interpeted data creating a SOLR index.
+
+All the steps generate an output. If only the final output is desired, the intermediate outputs can be ignored.
+
+Usage:
+  $CMD [options] dwca-avro     (<dr>...|all)
+  $CMD [options] interpret     (<dr>...|all) $WHEREL
+  $CMD [options] uuid          (<dr>...|all)         $WHERE
+  $CMD [options] export-latlng (<dr>...|all)         $WHERE
+  $CMD [options] sample        (<dr>...|all)
+  $CMD [options] sample-avro   (<dr>...|all)         $WHERE
+  $CMD [options] index         (<dr>...|all) $WHEREL
+  $CMD [options] do-all        (<dr>...|all) $WHEREL
+  $CMD -h | --help
+  $CMD -v | --version
+
+Options:
+  --config=<files>     Comma separated list of alternative la-pipeline yaml configurations (the last file has the highest precedence).
+  --extra-args=<args>  Additional "arg1=values,arg2=value" to pass to pipeline options (highest precedence than yml values).
+  --no-colors          No colorize logs output.
+  --dry-run            Print the commands without actually running them.
+  --debug              Debug $CMD.
+  -h --help            Show this help.
+  -v --version         Show version.
+----
+$CMD $VER
+Copyright (C) 2020 ALA Development Team
+License MPL v 1.1
```

Comment:
I notice ALA normally uses a different license (MPL) than GBIF (Apache).  We should probably have one license for the whole project.

Reply:
Apologies - I was looking at a stale version of the PR page and missed this comment.  I think we should remove the Copyright line and change the licence to Apache to comply with the existing licence.

Reply:
Sorry, my fault. I started this CLI prior to the GBIF merge and I use the MPL license here and there. @djtfmartin forget this.

Reply:
thanks @vjrj